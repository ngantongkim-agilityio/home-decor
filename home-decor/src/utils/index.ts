export * from './products';
export * from './notifications';
