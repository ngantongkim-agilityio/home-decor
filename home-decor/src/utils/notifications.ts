import * as Notifications from 'expo-notifications';

export const requestPermissionsNotifications = async () => {
  await Notifications.requestPermissionsAsync();
  Notifications.setNotificationHandler({
    handleNotification: async () => ({
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: false,
    }),
  });
};

export const scheduleNotification = async ({
  id = '',
  title = `You've got a new notification! ðŸ“¬`,
  body = 'Here is the notification message',
  seconds = 5,
}) =>
  await Notifications.scheduleNotificationAsync({
    content: { title, body, data: { url: `/details/${id}` } },
    trigger: {
      seconds,
      repeats: false,
    } as Notifications.TimeIntervalTriggerInput,
  });
