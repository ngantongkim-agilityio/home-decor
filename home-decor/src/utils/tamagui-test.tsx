import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { TamaguiProvider } from 'tamagui';
import { tamaguiConfig } from '@/themes/tamagui.config';

const Providers = ({ children }: PropsWithChildren) => {
  return <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>;
};

const baseRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';
export { baseRender as render };
