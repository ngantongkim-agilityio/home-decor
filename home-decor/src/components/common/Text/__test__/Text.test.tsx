import { render } from '@/utils/tamagui-test';
import { Text } from '..';

describe('Text Component', () => {
  it('Should render to match snapshot', () => {
    const component = render(<Text />);

    expect(component).toMatchSnapshot();
  });
});
