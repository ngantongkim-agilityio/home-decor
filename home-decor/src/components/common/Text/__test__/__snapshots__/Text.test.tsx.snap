// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component Should render to match snapshot 1`] = `
<Text
  style={
    {
      "color": {
        "dynamic": {
          "dark": "#f4b5a4",
          "light": "#f4b5a4",
        },
      },
      "fontFamily": "PoppinsRegular",
      "fontSize": "$4",
      "lineHeight": 17,
      "textDecorationLine": "none",
    }
  }
  suppressHighlighting={true}
/>
`;
